app/controllers/asset/noneeabusiness/remove/RemoveAssetEndDateController.scala

```diff
@@ -77,7 +79,11 @@ class RemoveAssetEndDateController @Inject()(
               Future.successful(BadRequest(view(formWithErrors, index, asset.orgName))),
             endDate => {
               trustService.removeAsset(request.userAnswers.identifier, RemoveAsset(AssetNameType.NonEeaBusinessAssetNameType, index, endDate)).map(_ =>
-                Redirect(controllers.asset.routes.AddAssetsController.onPageLoad())
+                if (request.userAnswers.isMigratingToTaxable) {
+                  Redirect(controllers.asset.nonTaxableToTaxable.routes.AddAssetsController.onPageLoad())
+                } else {
+                  Redirect(controllers.asset.noneeabusiness.routes.AddNonEeaBusinessAssetController.onPageLoad())
+                }
```

Comment:
Looks to be used twice in this controller as well. Could this be a method in the navigator?